src/metabase/lib/column_group.cljc

```diff
@@ -97,18 +94,7 @@
 (defmethod display-info-for-group-method :group-type/join.implicit
   [query stage-number {:keys [fk-field-id], :as _column-group}]
   (merge
-   (when-let [field (if (string? fk-field-id)
-                      ;; TODO: This is probably working around a QP bug. See #37067 - the `data.cols` from the API has
```

Comment:
Do we have a qp test that would produce a stringy `fk-field-id`? I'm worried that this will crop again.

Reply:
See the earlier PR #37079 - this logic was added speculatively trying to fix a stats-only issue. It didn't work, and this PR is the result. The string IDs were a symptom of the bug being fixed by this PR, so this logic can be removed now that the bug is properly fixed. There are tests for the real requirement here - FKs on SQL models.

Reply:
Understood